tfx/experimental/pipeline_testing/dummy_launcher_test.py

```diff
@@ -0,0 +1,153 @@
+# Lint as: python2, python3
+# Copyright 2019 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for tfx.components.trainer.executor."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import mock
+import tensorflow as tf
+from ml_metadata.proto import metadata_store_pb2
+from tensorflow.python.lib.io import file_io  # pylint: disable=g-direct-tensorflow-import
+from tfx.orchestration import data_types
+from tfx.orchestration import metadata
+from tfx.orchestration import publisher
+from tfx.experimental.pipeline_testing import dummy_component_launcher
+from tfx.experimental.pipeline_testing import dummy_executor
+from tfx.orchestration.launcher import test_utils
+from tfx.types import channel_utils
+
+class DummyLauncherTest(tf.test.TestCase):
+
+  @mock.patch.object(publisher, 'Publisher')
+  def testCustomDummyExecutor(self, mock_publisher):
+    # verify whether custom dummy substitution works
+    mock_publisher.return_value.publish_execution.return_value = {}
+
+    test_dir = os.path.join(
+        os.environ.get('TEST_UNDECLARED_OUTPUTS_DIR', self.get_temp_dir()),
+        self._testMethodName)
+
+    connection_config = metadata_store_pb2.ConnectionConfig()
+    connection_config.sqlite.SetInParent()
+    metadata_connection = metadata.Metadata(connection_config)
+
+    pipeline_root = os.path.join(test_dir, 'Test')
+    output_path = os.path.join(test_dir, 'output') # output/
+    record_path = os.environ['HOME'] # place holder
+
+    input_artifact = test_utils._InputArtifact() # pylint: disable=protected-access
```

Comment:
Super nit: Can you put two spaces in front of a comment? This is required in google internal linter.(Maybe we should add this rule if it is missing in our pylintrc.)